### PR TITLE
[BACKLOG-30667] CSRF- /api/scheduler changes

### PR DESCRIPTION
**Pull request by sVarma1992**
[BACKLOG-30667] - CSRF- /api/scheduler changes

### CODE REVIEW

File: plugins/engine-configuration/pom.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
   <properties>
     <pdi.version>9.3.0.0-SNAPSHOT</pdi.version>
+    <csrf.token.service.client.version>0.8.0</csrf.token.service.client.version>
```
Comment (dcleao) [inline plugins/engine-configuration/pom.xml:20]: There already exists a property in the [pentaho parent pom](https://github.com/pentaho/maven-parent-poms/blob/master/pom.xml#L353) for this that we _must_ use — `hv-security-web.version`.
So this line should be removed.

File: plugins/engine-configuration/impl/pom.xml [hunk 2]
```diff
@@ -77,6 +77,12 @@
       <groupId>org.apache.httpcomponents</groupId>
       <artifactId>httpcore</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.hitachivantara.security.web</groupId>
+      <artifactId>csrf-token-service-client-java-httpclient-v4</artifactId>
+      <version>${csrf.token.service.client.version}</version>
```
Comment (dcleao) [inline plugins/engine-configuration/impl/pom.xml:83]: Version must be inherited from the parent pom, so please remove the `version` element.

File: plugins/engine-configuration/assemblies/feature/src/main/feature/feature.xml [hunk 3]
```diff
@@ -26,6 +26,7 @@
 
   <feature name="${project.artifactId}" version="${project.version}">
     <details>${project.description}</details>
+    <bundle>mvn:com.hitachivantara.security.web/csrf-token-service-client-java-httpclient-v4/${csrf.token.service.client.version}</bundle>
```
Comment (dcleao) [inline plugins/engine-configuration/assemblies/feature/src/main/feature/feature.xml:29]: Must use the property `hv-security-web.version` inherited from the parent pom, instead.

**Review by dcleao — CHANGES_REQUESTED** [review]
Some POM issues

File: plugins/engine-configuration/pom.xml (hunk 1 above)
Comment (sVarma1992) [reply plugins/engine-configuration/pom.xml]: addressed the comments

File: plugins/engine-configuration/impl/pom.xml (hunk 2 above)
Comment (sVarma1992) [reply plugins/engine-configuration/impl/pom.xml]: addressed the comments

File: plugins/engine-configuration/assemblies/feature/src/main/feature/feature.xml (hunk 3 above)
Comment (sVarma1992) [reply plugins/engine-configuration/assemblies/feature/src/main/feature/feature.xml]: addressed the comments

File: plugins/engine-configuration/impl/pom.xml (hunk 2 above)
Comment (dcleao) [reply plugins/engine-configuration/impl/pom.xml]: You should remove the version element altogether.
> so please remove the version element.

**Review by dcleao — CHANGES_REQUESTED** [review]
Please see note.

File: plugins/engine-configuration/impl/pom.xml (hunk 2 above)
Comment (sVarma1992) [reply plugins/engine-configuration/impl/pom.xml]: Removed the version as requested

**Review by dcleao — APPROVED** [review]

**Pull request merged**